### PR TITLE
chore: revisit system data contracts

### PR DESCRIPTION
**Pull request by shumkov**
## Issue being fixed or feature implemented

We need to ensure that our system data contracts are up to date with the recent version of Drive and that there are no problems or mistakes.

## What was done?

Dashpay:
- Ensure that at least one of the profile fields is present (according to the DIP14)
- Ensure all three `avatarUrl`, `avatarHash` and `avatarFingerprint` are present or non of them 
- Switch to system `$createdAtCoreBlockHeight` instead of user-based

DPNS:
- Switched to mutability options instead of data triggers for `preorder`
- Allow future functionality for the `domain` documents but disable it temporary with data triggers so it will be easier to change in future.
- Added `$createdAt`, `$updatedAt` and `$transferedAt` for the `domain` document type

Withdrawals:
- Prevented creation of the withdrawal documents with document mutability options

## How Has This Been Tested?

- Updated dashpay contract tests
- Added additional tests for dashpay queries to validate indexes
- Added query tests for masternode reward shares

## Breaking Changes

Dashpay and DPNS contracts and their validation logic has changed so previously created blockchain data is not valid.

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have added "!" to the title and described breaking changes in the corresponding section if my code contains any
- [x] I have made corresponding changes to the documentation if needed

**For repository code-owners and collaborators only**
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

**Review by HashEngineering — COMMENTED** [review]
ACK

**Comment by QuantumExplorer** [comment]
We should add "$transferredAt" for DPNS domain

File: packages/dashpay-contract/schema/v1/dashpay.schema.json [hunk 1]
```diff
@@ -57,6 +57,12 @@
         "position": 4
       }
     },
+    "minProperties": 1,
+    "dependentRequired": {
+      "avatarUrl": ["avatarHash", "avatarFingerprint"],
```
Comment (shumkov) [inline packages/dashpay-contract/schema/v1/dashpay.schema.json:62]: We need `minLength`

File: packages/dpns-contract/schema/v1/dpns-contract-documents.json [hunk 2]
```diff
@@ -1,5 +1,7 @@
 {
   "domain": {
+    "documentsMutable": false,
```
Comment (shumkov) [inline packages/dpns-contract/schema/v1/dpns-contract-documents.json:3]: We should do it with data trigger so we can make them updatable easily in future.

Comment (QuantumExplorer) [inline packages/dpns-contract/schema/v1/dpns-contract-documents.json:3]: Use a data trigger here, and have mutable: true, canBeDeleted: true
Also add transferable: 1, tradeMode: 1

Then use data triggers for now to restrict everything.

File: packages/dpns-contract/schema/v1/dpns-contract-documents.json [hunk 3]
```diff
@@ -133,6 +136,7 @@
     "$comment": "In order to register a domain you need to create a preorder. The preorder step is needed to prevent man-in-the-middle attacks. normalizedLabel + '.' + normalizedParentDomain must not be longer than 253 chars length as defined by RFC 1035. Domain documents are immutable: modification and deletion are restricted"
   },
   "preorder": {
+    "documentsMutable": false,
```
Comment (QuantumExplorer) [inline packages/dpns-contract/schema/v1/dpns-contract-documents.json:139]: Needs "canBeDeleted": true to make it explicit.

File: packages/masternode-reward-shares-contract/schema/v1/masternode-reward-shares-documents.json [hunk 4]
```diff
@@ -14,14 +14,6 @@
...
         ],
         "unique": true
-      },
-      {
-        "name": "ownerId",
-        "properties": [
-          {
-            "$ownerId": "asc"
-          }
-        ]
```
Comment (QuantumExplorer) [inline packages/masternode-reward-shares-contract/schema/v1/masternode-reward-shares-documents.json:24]: Bring it back

**Review by QuantumExplorer — CHANGES_REQUESTED** [review]

File: packages/rs-drive-abci/src/execution/validation/state_transition/state_transitions/documents_batch/data_triggers/triggers/dashpay/v0/mod.rs [hunk 5]
```diff
@@ -50,82 +47,55 @@ pub(super) fn create_contact_request_data_trigger_v0(
...
 
-                return Ok(result);
-            }
-        }
+        return Ok(result);
     }
 
-    //  toUserId identity exits
-    let identity = context.platform.drive.fetch_identity_balance(
+    // TODO: Calculate fee operations
```
[the remark this answers is not in the document]
Comment (shumkov) [reply packages/rs-drive-abci/src/execution/validation/state_transition/state_transitions/documents_batch/data_triggers/triggers/dashpay/v0/mod.rs:89]: @QuantumExplorer this question for you

**Review by QuantumExplorer — DISMISSED** [review]

**Pull request merged**